scipy/stats/stats.py

```diff
@@ -7503,7 +7568,7 @@ def ranksums(x, y):
     s = np.sum(x, axis=0)
     expected = n1 * (n1+n2+1) / 2.0
     z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)
-    prob = 2 * distributions.norm.sf(abs(z))
+    z, prob = _normtest_finish(z, alternative)
```

Comment:
So in this function, which did not do something like
```python3
    if z.ndim == 0:
        z = z[()]
```
are we changing the behavior? I imagine that this is an unusual case; just thought I should check.

Reply:
> are we changing the behavior?

No, we aren't changing the behaviour here. It returned a `np.float64` for 1-D inputs before still does the same. Apparently, doing `z[()]` returns a `np.float64` for `np.float64` or `ndarray` `z` but `z.item()` returns a python float! Maybe, that's one of the reasons scipy functions prefer doing `z[()]` over `z.item()`.

Reply:
Thanks.

Reply:
Here's another reason: z[()] works on pretty much anything NumPy returns in just the way you'd want it to. If it's already a float, no problem. If it's a 0d array, it returns a float. If it's any other array, it just returns the array.